### ISSUE TITLE
Support jdbc:postgresql URL scheme for SQL shell

### ISSUE THREAD

**Issue by electrum** [cockroachdb/cockroach#63961]
When [Testcontainers](https://www.testcontainers.org/modules/databases/cockroachdb/) to start a Cockroach server for testing, it's helpful to grab the JDBC URL from the logs and connect from the shell:
```
INFO - Container is started (JDBC URL: jdbc:postgresql://localhost:55563/postgres)
```
This would be a bit more convenient if the shell could accept the `jdbc:postgresql` prefix. While the Cockroach URL format is not exactly the same as JDBC, it's close enough for this purpose, that simply chopping off the `jdbc:` works.

**Comment by andreimatei** [comment]
@knz what do you think of accepting this jdbc connection string syntax? Even if we don't accept it, perhaps the shell could print a hint or ask for a confirmation.

**Comment by knz** [comment]
> simply chopping off the jdbc: works.

is that right? IIRC, JDBC expects TLS client certs in PKCS#8 format, so the default URL spelled out by a server is likely not to work directly with JDBC.

Can someone spell out more clearly what's the difference between a jdbc url and a pg url?

**Comment by rafiss** [comment]
This comment has an enumeration: https://github.com/cockroachdb/cockroach/issues/63962#issuecomment-827240184

**Comment by knz** [comment]
Ok that other issue is very good. Why are we keeping 2 issues open?

**Comment by rafiss** [comment]
They are asking for different things
- #63961 (this one) is asking for `cockroach sql --url` to accept a URL that begins with `jdbc:`, so it's about which URLs we can parse.
- #63962 is asking for `cockroach demo` to output a URL that can be pasted directly into a Java program so it can connect. I'd imagine that `cockroach start` could also output such a URL. So it's about which URLs we display.

**Comment by knz** [comment]
I'd say for this present issue testcontainers needs to learn how to print out a regular crdb URL, not a JDBC one.

For the other issue, we should have server processes (incl `demo`) print both urls side-by-side, or perhaps ensure the base one is valid for jdbc.

**Comment by knz** [comment]
also, in the particular example at the top, the URL is incomplete - without a username and TLS certs, it's 99% likely that the `cockroach sql` shell won't have enough data in the URL to connect. So just stripping "jdbc:" in front is not going to be sufficient.

**Comment by electrum** [comment]
@knz you're right, I thought I had tested this, but it doesn't work. The JDBC driver does insecure by default, but the shell indeed requires `--insecure` or a URL parameter. Having Testcontainers print the `cockroach sql` command is a good idea; we've been adding that to our own programs that use it.

**Comment by electrum** [comment]
A minor correction on TLS client cert formats (something which I've had to deal with far too often):

- [PKCS #8](https://en.wikipedia.org/wiki/PKCS_8) is the typical PEM base64-encoded format that you see with OpenSSL and other UNIX tools, which is what `cockroach` uses.
- [PKCS #12](https://en.wikipedia.org/wiki/PKCS_12) is a binary format is now the default Java keystore format (to replace the custom JKS format).

For [Trino](https://trino.io/), which is written in Java, we did a bunch of extra work to make it support `PKCS #8 PEM` everywhere, which is typically much more convenient for users.
